gammapy/cube/tests/test_counts.py

```diff
@@ -82,3 +84,21 @@ def test_fill_map_counts_keyerror(events):
     cntmap = WcsNDMap.create(binsz=0.1, npix=10, axes=[axis])
     with pytest.raises(KeyError):
         fill_map_counts(cntmap, events)
+
+@requires_data('gammapy-extra')
+def test_fill_map_counts_multiple_energy_axes(events):
+    axis_mc = MapAxis([0, 2, 4], name='energy_mc', unit='TeV')
+    axis_reco = MapAxis([0.001, 1000.], name='energy_reco', unit='TeV')
+    axis = MapAxis([0.001, 1000.], name='energy', unit='TeV')
+
+    cntmap1 = WcsNDMap.create(binsz=1, npix=10, coordsys='GAL', axes=[axis, axis_mc])
+    cntmap2 = WcsNDMap.create(binsz=1, npix=10, coordsys='GAL', axes=[axis_mc, axis_reco])
+
+    col_mc = Column(np.ones(len(events.table)) * u.TeV, name='ENERGY_MC')
+    events.table.add_column(col_mc)
+
+    # Check that energy_mc entries are placed in the right axis
+    fill_map_counts(cntmap1, events)
+    assert_allclose(cntmap1.data.sum(axis=(1, 2, 3)), [105592, 0])
```

Comment:
I find this example hard to understand.

Like I said previously, I would much prefer a table with 3 or 5 events from scratch, and asserts on single well-chosen map pixels to see if they have a value of 0, 1 or 2, that is easy to understand from what is filled.

But OK, probably you've thought through this test case and it's a good one...

Reply:
Well, creating the correct eventlist from scratch is a bit annoying and it is not done so far in test_counts.py so I'll prefer to stick with the current one.

We are just checking that in the `energy_mc` axis on the map all entries are in bin 0 and that bin 1 is empty.